### DISCUSSION TITLE
[Feature request] Ability to scan codes from saved images

### DISCUSSION THREAD

**Discussion by vishnukvmd** [ente-io/ente#487] [Enhancements]
Some apps might not provide a way to copy-paste codes, in which case it would be useful to take a screenshot of the code, and add it to Auth by loading the saved screenshot.

Should ideally warn the user to delete the saved screenshot post addition.

**Comment by sken130** [comment]
Yes, this feature is useful for migrating from Google Authenticator to ente Auth if the user doesn't have an extra smartphone nor a laptop with a camera.

**Comment by sken130** [comment]
Though, I suggest this app should remind user to pause all cloud photo backup/sync apps in the smartphone, before taking any screenshot of QR code from other authenticator apps, to avoid the secret QR code ever leaving the device.

In summary, the suggested implementation is:

- In Settings > Data > Import codes, add a new option "Google Authenticator (saved image)"
- When user presses this option, remind user to turn off all photo cloud sync from all apps, including iCloud, Google Photo, OneDrive, etc. Also remind user that if they have a second smartphone, it is safer to import by scanning QR code.
- After import, remind user to delete the screenshot before resuming any photo cloud sync. Or perhaps confirm to help user delete the screenshot

**Comment by i-aiymen** [comment]
Added a PR for this issue. Do check it out

https://github.com/ente-io/auth/pull/336
